packages/preferences/src/common/preference-service.ts

```diff
@@ -32,60 +34,70 @@ export class PreferenceService implements Disposable {
         this.toDispose.dispose();
     }
 
+    ready(): Promise<void> {
+        return this.isReady;
+    }
+
     protected onDidChangePreference(event: PreferenceChangedEvent): void {
+        for (const prefChange of event.changes) {
+            // Pref removed
+            if (prefChange.oldValue !== undefined && prefChange.newValue === undefined) {
```

Comment:
`if(change.newValue === undefined || change.newValue === null)` should do, it is not important whether there is `oldValue`